docs/dev_building_web_assembly.rst

```diff
@@ -0,0 +1,146 @@
+
+=========================================
+Building libCellML for use in the Browser
+=========================================
+
+To build libCellML for the browser we use the emscripten toolchain to create web assembly binaries.
+Here we will go through the process of building web assembly binaries on macOS.
+For other operating systems modifications to these instructions will have to be made.
+Any modifications required are expected to be self evident – with this in mind these instructions should be universally applicable.
+
+Preparation
+===========
+
+To build libCellML for the browser we will need the following tools ready for use.
+
+- Emscripten
+- Compiler toolchain
+- CMake (at least version 3.14)
+- Git
+
+Building
+========
+
+libCellML requires two dependencies to be built before libCellML itself can be built.
+We will clone and build those first and then move on to building libCellML.
+
+Dependencies
+------------
+
+Start by creating a directory under which all our files will live and make it our current directory for subsequent commands::
+
+  mkdir emscripten
+  cd emscripten
+
+Now we can clone our dependencies (we suggest using these exact repositories other repositories may work but these repositories are known to work)::
+
+  git clone https://github.com/OpenCMISS-Dependencies/zlib.git -b v1.2.3
+  git clone https://github.com/OpenCMISS-Dependencies/libxml2.git -b v2.9.10
+
+We will create two directories to hold the build files::
+
+  mkdir build-zlib-release
+  mkdir build-libxml2-release
+
+Then we will configure, build, and install zlib as this is required by libxml2.
+We will install zlib under our emscripten working directory and **not** into the system directories::
+
+  emcmake cmake -S zlib -B build-zlib-release -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=usr/local
+
+Here we have configured the project using the emscripten wrapper script `emcmake`.
+This script sets up environment variables for us to make building with the emscripten compilers easy.
+We also configure the zlib library to be built in `Release` mode and set the install prefix relative to the **current** directory.
+However this is not an issue for us as the **source** and **build** directories are sibling directories.
+
+To build and install the zlib library we simply issue these commands::
+
+  cd build-zlib-release
+  make
+  make install
+
+Next we follow a similar procedure for building `libxml2`.
+Start by making the current directory the parent of the zlib build directory::
+
+  cd ..
+
+Then as before configure the library::
+
+  emcmake cmake -S libxml2 -B build-libxml2-release -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=usr/local -DZLIB_DIR=../usr/local/lib/cmake/ZLIB/ -DCMAKE_PREFIX_PATH=usr/local -DBUILD_SHARED_LIBS=OFF -DLIBXML2_WITH_ICONV=OFF -DLIBXML2_WITH_LZMA=OFF -DLIBXML2_WITH_PYTHON=OFF -DLIBXML2_WITH_TESTS=OFF -DLIBXML2_WITH_PROGRAMS=OFF
+
+There are seven additions to the configure command this time;
```

Comment:
There are seven additions to the configure command this time::

Reply:
9edec02